zerver/models.py

```diff
@@ -736,7 +736,10 @@ class UserProfile(AbstractBaseUser, PermissionsMixin):
     MAX_NAME_LENGTH = 100
     MIN_NAME_LENGTH = 2
     API_KEY_LENGTH = 32
-    NAME_INVALID_CHARS = ['*', '`', '>', '"', '@']
+    NAME_INVALID_CHARS = ['*', '`', '"', '@',
```

Comment:
I think the hardest part of this issue is making a defensible argument for this list. Both why this list is sufficient to satisfy the RFC, and why it can't be much shorter.

E.g. we'd like to allow several of these (like `(` and `)`) if we can.

In the best case there's some built-in function in python or django that handles the validation for us. Do you know if there might be such a function?

Reply:
Upon further reading, I may have misunderstood RFC 5322 slightly. There seems to be a seperate class besides the atext wherein all ascii codes excluding `\` and `"` are allowed. We could possibly automate handling quoting the text if needed through using `email.util.parseaddr()`. 

I am going to check the behavior of `parseaddr()` against different inputs and if it works well, it might be best to use parseaddr and drop the additional strength changes completely.

Reply:
The specific part of RFC 5322 that is relevant to this would be page 13, for reference.